src/mono/nuget/Microsoft.NET.Runtime.RuntimeConfigParser.Task/Sdk/Sdk.props

```
@@ -1,6 +1,7 @@
 <Project>
   <PropertyGroup>
-    <RuntimeConfigParserTasksAssemblyPath>$(MSBuildThisFileDirectory)..\tasks\RuntimeConfigParser.dll</RuntimeConfigParserTasksAssemblyPath>
+    <RuntimeConfigParserTasksAssemblyPath Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\tasks\net6.0\RuntimeConfigParser.dll</RuntimeConfigParserTasksAssemblyPath>
+    <RuntimeConfigParserTasksAssemblyPath Condition="'$(MSBuildRuntimeType)' != 'Core'">$(MSBuildThisFileDirectory)..\tasks\net472\RuntimeConfigParser.dll</RuntimeConfigParserTasksAssemblyPath>
```

Not sure if it's available here but this should use $(NetCoreAppToolCurrent) and $(TargetFrameworkForNETFramework) since this is how we're building the RuntimeConfigParser.csproj

Although I wonder if we should change that to hardcode a specific version since NetCoreAppToolCurrent was more designed for tasks used inside of the dotnet/runtime repo and we're shipping more tasks outside of it now:

```suggestion
    <RuntimeConfigParserTasksAssemblyPath Condition="'$(MSBuildRuntimeType)' == 'Core'">$(MSBuildThisFileDirectory)..\tasks\$(NetCoreAppToolCurrent)\RuntimeConfigParser.dll</RuntimeConfigParserTasksAssemblyPath>
    <RuntimeConfigParserTasksAssemblyPath Condition="'$(MSBuildRuntimeType)' != 'Core'">$(MSBuildThisFileDirectory)..\tasks\$(TargetFrameworkForNETFramework)\RuntimeConfigParser.dll</RuntimeConfigParserTasksAssemblyPath>
```

---

Actually, scratch that. The Sdk.props is imported on the user's machine so we won't have $(NetCoreAppToolCurrent).

---

The gist of my comment still remains though: we need to make sure we either update this place here whenever we bump the `NetCoreAppToolCurrent` or stop using `NetCoreAppToolCurrent` in the tasks csproj

---

Created a follow up task https://github.com/dotnet/runtime/issues/53873